addons/hr_recruitment_skills/models/hr_applicant.py

```diff
@@ -158,9 +158,4 @@ def write(self, vals):
             original_vals = vals.copy()
             original_vals["applicant_skill_ids"] = skills
             vals["applicant_skill_ids"] = self.env["hr.applicant.skill"]._get_transformed_commands(skills, self)
-            for applicant in self:
-                # Modify the skill values for the talent if it exists
-                if applicant.pool_applicant_id and (not applicant.is_pool_applicant):
-                    mapped_skills = applicant._map_applicant_skill_ids_to_talent_skill_ids(original_vals)
```

Comment:
This method `_map_applicant_skill_ids_to_talent_skill_ids` looks sophisticated so I fear your simple way of updating skills may not be enough. Could you double-check?

Note that this `_map_applicant_skill_ids_to_talent_skill_ids` is only used here, so you may want to remove the entire method definition as well if you end up with the conclusion it is not needed